src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+        }
+        deleteDrivers(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement preparedStatement =
+                            connection.prepareStatement(deleteRequest)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete a driver by id " + id + " ",
+                    throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest
+                = "SELECT * FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN drivers d ON d.id = cd.driver_id WHERE cd.driver_id = ? "
+                + "AND c.is_deleted = FALSE AND d.is_deleted = FALSE;";
+        List<Car> carsList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(
+                        getAllByDriverRequest)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carsList.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Couldn't get a list of cars by driver Id "
+                    + driverId, throwables);
+        }
+        carsList.forEach(car -> car.setDriverList(getDriversForCar(car.getId())));
+        return carsList;
+    }
+
+    private Car parseCarWithManufacturerFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setModel(resultSet.getString("model"));
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        car.setManufacturer(manufacturer);
+        car.setId(resultSet.getObject("car_id", Long.class));
+        return car;
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriversRequest = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addDriverToCarStatement =
+                        connection.prepareStatement(insertDriversRequest,
+                             Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
You don't use Statement.RETURN_GENERATED_KEYS anywhere on this method :)

Reply:
done